src/geometry/cpu/geometry_op_impl.cc

```diff
@@ -4,14 +4,29 @@
  * \brief Geometry operator CPU implementation
```

Comment:
I suggest merging `src/geometry` into `src/array/kernel`, @hetong007 WDYT?

Reply:
I agree that `src/geometry` is not rich enough to be separated. But is `src/array` the best place to hold it? I found it is mostly about scatter and matrix multiplication.